curriculum/challenges/english/06-quality-assurance/quality-assurance-projects/metric-imperial-converter.md

```diff
@@ -85,33 +85,128 @@ tests:
+        assert.equal(data1.initUnit, 'gal');
+        assert.equal(data1.returnUnit, 'L');
+        const data2 = await $.get(getUserInput('url') + '/api/convert?input=10L');
+        assert.equal(data2.initUnit, 'L');
+        assert.equal(data2.returnUnit, 'gal');
+        const data3 = await $.get(getUserInput('url') + '/api/convert?input=1l');
+        assert.equal(data3.initUnit, 'L');
+        assert.equal(data3.returnUnit, 'gal');
+        const data4 = await $.get(getUserInput('url') + '/api/convert?input=10KM');
+        assert.equal(data4.initUnit, 'km');
+        assert.equal(data4.returnUnit, 'mi');
+      } catch(xhr) {
+        throw new Error(xhr.responseText || xhr.message);
+      }
+    }"
   - text: If my unit of measurement is invalid, returned will be <code>'invalid unit'</code>.
     testString: "async getUserInput => { 
       try {
         const data = await $.get(getUserInput('url') + '/api/convert?input=1min');
-        assert(data.initUnit === 'invalid unit' || data === 'invalid unit');
+        assert(data.error === 'invalid unit' || data === 'invalid unit');
       } catch(xhr) {
-        throw new Error(xhr.responseText);
+        throw new Error(xhr.responseText || xhr.message);
       }
     }
     "
-  - text: If my number is invalid, returned with will 'invalid number'.
-    testString: ''
-  - text: If both are invalid, return will be 'invalid number and unit'.
-    testString: ''
+  - text: If my number is invalid, returned will be <code>'invalid number'</code>.
+    testString: "async getUserInput => { 
+      try {
+        const data = await $.get(getUserInput('url') + '/api/convert?input=1//2gal');
+        assert(data.error === 'invalid number' || data === 'invalid number');
+      } catch(xhr) {
+        throw new Error(xhr.responseText || xhr.message);
+      }
+    }"
+  - text: If both the unit and number are invalid, returned will be <code>'invalid number and unit'</code>.
+    testString: "async getUserInput => { 
+      try {
+        const data = await $.get(getUserInput('url') + '/api/convert?input=1//2min');
+        assert(data.error === 'invalid number and unit' || data === 'invalid number and unit');
+      } catch(xhr) {
+        throw new Error(xhr.responseText || xhr.message);
+      }
+    }"
   - text: I can use fractions, decimals or both in my parameter(ie. 5, 1/2, 2.5/6), but if nothing is provided it will default to 1.
-    testString: ''
-  - text: My return will consist of the initNum, initUnit, returnNum, returnUnit, and string spelling out units in format '{initNum} {initial_Units} converts to {returnNum} {return_Units}' with the result rounded to 5 decimals in the string.
-    testString: ''
+    testString: "async getUserInput => {
+      try {
+        const data1 = await $.get(getUserInput('url') + '/api/convert?input=1/2mi');
+        assert.approximately(data1.initNum, 1/2, 0.001);
+        assert.approximately(data1.returnNum, 0.80467, 0.001);
```

Comment:
```suggestion
        const data1 = await $.get(getUserInput('url') + '/api/convert?input=mi');
        assert.approximately(data1.initNum, 1, 0.001);
        assert.approximately(data1.returnNum, 1.60934, 0.001);
```

One test for "if nothing is provided it will default to 1"